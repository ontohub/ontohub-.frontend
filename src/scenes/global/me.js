import React from "react";
import gql from "graphql-tag";
import { Query } from "react-apollo";

export const MeContext = React.createContext();

export const query = gql`
  query CurrentUser {
    me {
      id
      displayName
      email
      emailHash
      __typename
    }
  }
`;

export const MeProvider = ({ children }) => (
  <Query query={query}>
    {({ data, loading, error }) => {
      if (error || loading) return children;

      return (
        <MeContext.Provider value={!loading && !error && data.me}>
          {children}
        </MeContext.Provider>
      );
    }}
  </Query>
);
