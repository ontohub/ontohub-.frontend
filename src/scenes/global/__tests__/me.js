import { MeContext, MeProvider, query } from "../me";

describe("MeProvider", () => {
  set("component", () =>
    render(
      <MeProvider>
        <MeContext.Consumer>
          {me => (
            <React.Fragment>
              <h1>{me && me.id}</h1>
              <h2>{me && me.emailHash}</h2>
            </React.Fragment>
          )}
        </MeContext.Consumer>
      </MeProvider>
    )
  );

  describe("no data loaded", () => {
    set("mockedResponse", () => ({
      loading: true
    }));

    it("matches the snapshot", () => {
      mockResponse(query, mockedResponse);
      expect(component).toMatchSnapshot();
    });
  });

  describe("Loaded data", () => {
    set("mockedResponse", () => ({
      data: {
        me: { id: "ada", emailHash: "14758f1afd44c09b7992073ccf00b43d" }
      }
    }));

    it("matches the snapshot", () => {
      mockResponse(query, mockedResponse);
      expect(component).toMatchSnapshot();
    });
  });
});
