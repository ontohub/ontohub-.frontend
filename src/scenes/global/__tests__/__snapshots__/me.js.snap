// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeProvider Loaded data matches the snapshot 1`] = `
Array [
  <h1>
    ada
</h1>,
  <h2>
    14758f1afd44c09b7992073ccf00b43d
</h2>,
]
`;

exports[`MeProvider no data loaded matches the snapshot 1`] = `
Array [
  <h1>
    
</h1>,
  <h2>
    
</h2>,
]
`;
